examples/Makefile.testing

```diff
@@ -0,0 +1,211 @@
+.PHONY:
+
+# This makefile is for creating clusters filled with repos and pipelines for end-to-end and functional testing.
+# It builds on targets in the main Makefile.  You would invoke it by running the
+# "cd <source-location>/examples  ; make -s -f Makefile.testing dirty-testing-cluster" commands
+# That target, "dirty-testing-cluster" is the first workhorse testing target that's been developed.
+# It will create a cluster that has failed jobs, failed commmits, and deleted commits.
+# Invoking it with "-s" will produce terse, informative output on the steps it's taking to create the
+# cluster as it's working.  You might want to run a "watch -n 2 kubectl get pods" and a
+# "watch -n 2 pachctl list job" in other windows so that you can understand what's happening as it runs.
+# At the end, it will print a message telling you the cluster is ready for your testing, and you
+# can use it to test extract/restore or for whatever other testing purposes you want.
+# There are some top-level functions defined, like "pipeline-status" and "wait-for-jobs", that make it easier
+# to write targets that methodically create a cluster that's useful for testing.
+# The executable paths below are set to generic values.
+# Modify them for your system by setting environment variables, using a command like this
+# to both fetch the version of pachyderm you want to test and execute it.
+# "../etc/fetch_release_pachctl.py 1.10.0 ; env PACHCTL=${GOPATH}/bin/pachctl make -e  -f Makefile.testing dirty-testing-cluster)"
+
+include Makefile
+
+CP := cp
+DOCKER := docker
+ECHO := echo
+JQ := jq
+JQFLAGS :=
+KUBECTL := kubectl
+PACHCTL := pachctl
+RM := rm
+SLEEP := sleep
+SORT := sort
+TOUCH := touch
+UNIQ  := uniq
+WC := wc
```

Comment:
what's the utility of having override-able variables for standard unix tools?

Reply:
There are sometimes subtle differences across platforms on the standard tools. For example, "echo" on macOS doesn't have the -e flag.  When I build these kind of files, I like the ability to call out the particular executables that'll be called, just in case.

Reply:
I've broken the complicated makefile functions out into what should be more understandable and reusable scripts in etc/examples.

I've removed most of these, leaving only those that have actual functional differences that could affect the running of the new scripts in etc/examples.